### PR TITLE
Added GVDDK support for the QueryAllocatedBlocks function.

### PR DESCRIPTION
**Pull request by carlbraganza**
Made the GVDDK unit tests work for VDDK 7.0 and support test invocation from under bash.

### CODE REVIEW

**Comment by carlbraganza** [comment]
In the unit tests I replaced use of the environment variable `PWD` with `PASSWORD` as `PWD` is a built-in in `bash`.  Please let me know if I should support both - i.e. fall back to `PWD` if `PASSWORD` is not specified.

File: vendor/github.com/vmware/gvddk/gvddk-high/gvddk.go [hunk 1]
```diff
@@ -330,4 +330,9 @@ func (this DiskConnectHandle) Close() error {
 
 func (this DiskConnectHandle) Capacity() int64 {
 	return int64(this.info.Capacity) * gDiskLib.VIXDISKLIB_SECTOR_SIZE
-}
\ No newline at end of file
+}
+
+// QueryAllocatedBlocks invokes the VDDK function of the same name.
+func (this DiskReaderWriter) QueryAllocatedBlocks(startSector gDiskLib.VixDiskLibSectorType, numSectors gDiskLib.VixDiskLibSectorType, chunkSize gDiskLib.VixDiskLibSectorType) ([]gDiskLib.VixDiskLibBlock, gDiskLib.VddkError) {
```
Comment (dsu-igeek) [inline vendor/github.com/vmware/gvddk/gvddk-high/gvddk.go:336]: Sorry for the late review.  Can we move this to DiskConnectHandle?  It doesn't seem like a Reader/Writer method.

**Review by dsu-igeek — CHANGES_REQUESTED** [review]
Looks good, just one thought on which high level object the API should be on.  Thanks!

File: vendor/github.com/vmware/gvddk/gvddk-high/gvddk.go (hunk 1 above)
Comment (carlbraganza) [reply vendor/github.com/vmware/gvddk/gvddk-high/gvddk.go:336]: Thanks, what you say make sense! 

Would you object, though, if it still gets exposed via `DiskReaderWriter` along the lines of the `DiskReaderWrite.ReadAt()` and `DiskReaderWrite.WriteAt()` methods? e.g.
```
func (this DiskReaderWriter) QueryAllocatedBlocks(startSector gDiskLib.VixDiskLibSectorType, numSectors gDiskLib.VixDiskLibSectorType, chunkSize gDiskLib.VixDiskLibSectorType) ([]gDiskLib.VixDiskLibBlock, gDiskLib.VddkError) {
	return this.diskHandle.QueryAllocatedBlocks(startSector, numSectors, chunkSize)
}
```

The reason I ask is that the `gvddk_high.Open` method calls many `gDiskLib` methods internally, including `PrepareForAccess`. Besides being super-convenient, I don't know if one *can* open another handle on the same volume to just make the `QueryAllocatedBlocks` call.

**Comment by xing-yang** [comment]
Can you add a line or two to explain why this change is needed in the PR description and in the release notes?  For example, it can accelerate the backup process by knowing which blocks are allocated.  Are you using this for incremental backups?

**Comment by carlbraganza** [comment]
> Can you add a line or two to explain why this change is needed in the PR description and in the release notes? For example, it can accelerate the backup process by knowing which blocks are allocated. Are you using this for incremental backups?

Will do. Just FYI, in my (admittedly limited) experience with vSphere, I found it most useful in **full backups** of FCDs, as  block devices with filesystems on them are relatively sparse initially. It is of limited use when change block tracking is enabled because the `QueryDiskChangedAreas` function returns data that is either identical or a subset of that returned by the `QueryAllocatedBlocks` function - the [VDDK Programming Guide ](https://code.vmware.com/docs/11750/virtual-disk-development-kit-programming-guide/GUID-152B7BA2-5A08-4044-A0FB-8DAACB39C4E1.html) describes the use of this method and how to use it.

File: vendor/github.com/vmware/gvddk/gvddk-high/gvddk.go (hunk 1 above)
Comment (dsu-igeek) [reply vendor/github.com/vmware/gvddk/gvddk-high/gvddk.go:336]: Sure, we can add it to both, thanks

**Review by dsu-igeek — APPROVED** [review]
Looks good to me

**Pull request merged**